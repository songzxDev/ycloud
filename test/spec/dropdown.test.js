import ko from 'knockout'
import 'ko-bindinghandler'
import comp from '../../src/components/dropdown/index.js'
import $ from 'jquery'
describe('y-dropdown组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + comp.name, {
    viewModel: comp.init,
    template: comp.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    title: ko.observable('test'),
    item: {value: 1,label:'男'},
    dataList: ko.observableArray([{value: 1, label: 'label1'}, {value: 2, label: 'label2'}]),
    isShow: ko.observable(true)
  }
  document.body.innerHTML = '<y-dropdown params="width:300, isShow:isShow, isStopTransferDom: true"><div data-bind="foreach:$parent.dataList">' +
      '<div class="mytestClass" data-bind="text:$data.label"></div>'
    '</div></y-dropdown>'
  ko.applyBindings(vm)
  var el = $('body')
  it("组件注册测试", () => {
    expect(comp).not.toBeNull();
    expect(ko.components.isRegistered(PREFIX + comp.name)).toBe(true);
  })
  it('验证width参数', () => {
    expect($('body').find('.y-dropdown').css('width')).toEqual('300px')
  })
  it("测试数据内容多次渲染", () => {
    var el = $('body').find('.mytestClass')
    var value = el.eq(0).text().trim()
    expect(value).toEqual(vm.dataList()[0].label)
    var value2 = el.eq(1).text().trim()
    expect(value2).toEqual(vm.dataList()[1].label)
  })
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
