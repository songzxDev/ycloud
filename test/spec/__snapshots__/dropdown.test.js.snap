// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-dropdown组件 快照 1`] = `
<body>
  <y-dropdown
    params="width:300, isShow:isShow, isStopTransferDom: true"
  >
    <div
      class="y-dropdown v-transfer-dom slideInUp"
      data-bind="
  transferdom: isTransferDom,
  style: style,
  css: {
    animated: animated,
    slideInUp: isShow,
    slideOutUp: !isShow(),
    hidden:!isShow()
  }"
      style="width: 300px; position: absolute;"
    >
      
  
      <!-- ko template: { nodes: $componentTemplateNodes } -->
      <div
        data-bind="foreach:$parent.dataList"
      >
        <div
          class="mytestClass"
          data-bind="text:$data.label"
        >
          label1
        </div>
        <div
          class="mytestClass"
          data-bind="text:$data.label"
        >
          label2
        </div>
      </div>
      <!-- /ko -->
      

    </div>
    

  </y-dropdown>
</body>
`;
